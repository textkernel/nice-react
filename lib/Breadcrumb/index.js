export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';