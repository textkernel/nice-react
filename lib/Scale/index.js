export { default as ScaleGroup } from './ScaleGroup';
export { default as ScaleItem } from './ScaleItem';