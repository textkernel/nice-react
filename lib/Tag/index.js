export { default as Tags } from './Tags';
export { default as Tag } from './Tag';