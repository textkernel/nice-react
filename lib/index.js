import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';