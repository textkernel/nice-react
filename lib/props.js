const contexts = ['brand', 'link', 'primary', 'accent', 'good', 'warning', 'bad'];
const sizes = ['small', 'medium', 'large'];
 
const props = {
  contexts,
  buttonContexts: contexts,
  buttonSizes: sizes 
};

module.exports = props;