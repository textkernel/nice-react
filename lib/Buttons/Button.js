import React, { Component } from 'react';
import PropTypes from 'prop-types';
import props from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    const { children, icon, alternative } = this.props;
    const iconClass = icon ? `icon ${icon}` : null;

    const regularButton = () => {
      const { context, size } = this.props;
      const baseClass = 'nice-btn';
      return [baseClass, context, size, iconClass];
    };

    const alternativeButton = () => {
      const baseClass = 'nice-btn-alt';
      return [baseClass, iconClass];
    };

    const butonClasses = alternative ? alternativeButton : regularButton;
    const classes = butonClasses().filter(c => c && c.length > 0);

    return (
      <a href='#' onClick={e => this.handleClick(e)} className={classes.join(' ')}>
        {children}
      </a>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(props.buttonContexts),
  size: PropTypes.oneOf(props.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  alternative: PropTypes.bool
};

Button.defaultProps = {
  context: 'default',
  size: 'medium',
  children: null,
  onClick: null,
  icon: null,
  alternative: false
};