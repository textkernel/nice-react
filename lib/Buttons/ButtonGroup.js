import React from 'react';
import PropTypes from 'prop-types';
import props from '../props';

const ButtonGroup = ({ children, position, size }) => {
  const classes = ['nice-btn-group', position, size].filter(c => c && c.length > 0);

  return (
    <div className={classes}>
      {children}
    </div>
  );
};

ButtonGroup.propTypes = {
  children: PropTypes.node,
  size: PropTypes.oneOf(props.buttonGroupSizes),
  position: PropTypes.string,
};

ButtonGroup.defaultProps = {
  size: 'small',
  children: null,
  position: 'justified' 
};

export default ButtonGroup;