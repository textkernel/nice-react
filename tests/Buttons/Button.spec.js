import React from 'react';
import { Button } from '../../lib/Buttons';

describe('<Button />', () => {
  it('should render the Button', () => {
    const wrapper = shallow(<Button>Default</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});