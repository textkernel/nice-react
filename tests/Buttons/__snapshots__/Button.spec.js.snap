// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render size Button 1`] = `
<a
  className="nice-btn primary large"
  href="#"
  onClick={[Function]}
>
  Large
</a>
`;

exports[`<Button /> should render style Button 1`] = `
<a
  className="nice-btn primary medium"
  href="#"
  onClick={[Function]}
>
  Primary
</a>
`;

exports[`<Button /> should render style and size Button 1`] = `
<a
  className="nice-btn primary large"
  href="#"
  onClick={[Function]}
>
  Primary large
</a>
`;

exports[`<Button /> should render the Button 1`] = `
<a
  className="nice-btn primary medium"
  href="#"
  onClick={[Function]}
>
  Default
</a>
`;
